relay-general/src/protocol/span.rs

```diff
@@ -59,6 +69,29 @@ pub struct Span {
     pub other: Object<Value>,
 }
 
+impl From<&Event> for Span {
+    fn from(event: &Event) -> Self {
+        let mut span = Self {
+            timestamp: event.timestamp.clone(),
+            start_timestamp: event.start_timestamp.clone(),
+            is_segment: Some(true).into(),
+            ..Default::default()
+        };
+
+        if let Some(trace_context) = event.context::<TraceContext>().cloned() {
```

Comment:
Even with this condition, an error event will also be converted into a span. Should we use [`TryFrom`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) instead of `From`, and error if the event isn't a transaction?

Reply:
I considered that, but then thought it's fine to leave this responsibility to the caller, since there is no formal reason why a conversion from error to span should fail. It might be a good safeguard though, let me reconsider.

Reply:
> since there is no formal reason why a conversion from error to span should fail

Shouldn't that case be disallowed, by the definition of an error? I think leaving this responsibility to the caller is okay, but it makes it easier to introduce bugs IMO. I'll let you decide.